app/controllers/books_controller.rb

```diff
@@ -108,4 +110,16 @@ def require_admin
     flash[:danger] = t "only_admin"
     redirect_to books_path
   end
+
+  def rate
+    sum = 0
+    if @book.reviews.empty?
+      @rate = nil
+    else
+      @book.reviews.each do |review|
+        sum += review.rate
+      end
```

Comment:
```suggestion
      @book.reviews.pluck(:rate).sum
```
Thử như này xem ok không em